### ISSUE TITLE
How to update defaultTime?

### ISSUE THREAD

**Issue by berstend** [dmtrKovalenko/material-ui-datetimepicker#18]
Hi there,
thanks for this very nice component!

I ran into a weird problem I can't figure out on my own:

I'm trying to update the defaultTime after DateTimePicker is mounted but changing the assigned state value is not re-rendering the component properly:

```es6
          <DateTimePicker
            defaultTime={this.state.dateTime}

```

When e.g. running this in a click handler the input field of DateTimerPicker is not updated:
```es6
this.setState({dateTime: moment().add(30, "minutes")})
```

Is there a way to update the dateTime from the parent component or is this currently not supported?

Thanks!

**Comment by berstend** [comment]
I've made a quick self-contained demo to show what I mean:
https://codesandbox.io/s/n91wn3vzm0

```es6
import React from 'react';
import { render } from 'react-dom';

import moment from 'moment';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import DateTimePicker from 'material-ui-datetimepicker';
import DatePickerDialog from 'material-ui/DatePicker/DatePickerDialog'
import TimePickerDialog from 'material-ui/TimePicker/TimePickerDialog';

class Demo extends React.Component {
  state = {
    dateTime: moment()
  }

  setDate(dateTime) {
    this.setState({ dateTime })
  }

  updateTime() {
    this.setState({ dateTime: moment().add(30, "minutes")})
    alert("Time updated")
  }

  render() {
    return (
      <div>
        <MuiThemeProvider>
          <DateTimePicker
            onChange={this.setDate.bind(this)}
            defaultTime={this.state.dateTime}
            DatePicker={DatePickerDialog}
            TimePicker={TimePickerDialog}
          />
        </MuiThemeProvider>
        <button onClick={this.updateTime.bind(this)}>Add 30 minutes</button>
      </div>
    );
  }
}

render(<Demo />, document.getElementById('root'));
```

**Comment by berstend** [comment]
I've hacked around this issue for now by abusing conditional rendering 😅:

```es6

  forceRerender() {
    this.setState({showPicker: false})
    setTimeout(() => {
      this.setState({showPicker: true})
    }, 0)
  }


// within render
          { this.state.showPicker && <DateTimePicker
            onChange={this.setDate.bind(this)}
            format='MMM DD, YYYY HH:mm'
            timePickerDelay={50}
            // ...
          />
          }
```

**Comment by dmtrKovalenko** [comment]
Hi, seems that I should fix that from the picker side. 
Will do that tomorrow and publish the release.

**Comment by dmtrKovalenko** [comment]
Fixed in 1.0.3, I have add ability to change state outside via `value` prop.

**Comment by berstend** [comment]
Amazing, just in time for my app release. 😄 

I've just updated and can confirm that setting the `value` prop will update the Picker as expected.

Thanks a lot and especially for fixing this so fast! 👍 

Kudos
